internal/pkg/term/selector/local_selector.go

```diff
@@ -145,6 +145,61 @@ func NewLocalFileSelector(prompt Prompter, fs afero.Fs) (*localFileSelector, err
 	}, nil
 }
 
+// StaticSources asks the user to select from a list of directories and files in the current directory and two levels down.
+func (s *localFileSelector) StaticSources(selPrompt, selHelp, anotherPathPrompt, anotherPathHelp string, pathValidator prompt.ValidatorFunc) ([]string, error) {
+	dirsAndFiles, err := s.listDirsAndFiles()
+	if err != nil {
+		return nil, err
+	}
+	if len(dirsAndFiles) == 0 {
+		log.Warningln("No directories or files were found in the current working directory. Enter a relative path with the 'custom path' option, or run this command from higher up in your local file directory.")
+	}
+	dirsAndFiles = append(dirsAndFiles, []string{staticSourceUseCustomPrompt}...)
+	var selections []string
+	selections, err = s.prompt.MultiSelect(
+		selPrompt,
+		selHelp,
+		dirsAndFiles,
+		func(v interface{}) error {
+			return nil
+		},
+		prompt.WithFinalMessage(staticAssetsFinalMsg),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("select directories and/or files: %w", err)
+	}
+	for i, selection := range selections {
+		anotherCustomPath := true
+		if selection == staticSourceUseCustomPrompt {
+			for anotherCustomPath {
+				customPath, err := s.prompt.Get(
+					anotherPathPrompt,
+					anotherPathHelp,
+					pathValidator,
+					prompt.WithFinalMessage(customPathFinalMsg))
+				if err != nil {
+					return nil, fmt.Errorf("get custom directory or file path: %w", err)
+				}
+				if selection == staticSourceUseCustomPrompt {
+					selections[i] = customPath // The first custom path replaces the prompt string.
+					selection = customPath
+				} else {
+					selections = append(selections, customPath) // Subsequent custom paths are appended.
+				}
```

Comment:
The way we are manipulating & examining `selection`  is not super straightforward I think 🤔  I have an alternative proposal to the codes here 💭  not a huge fan of ⬇️ either, but sharing anyway!
```go
addCustomIndex := -1
for i, selection := range selections {
  if selection == statisSourceUseCustomPrompt {
    addCustomIndex = i
    break
  }
}

 if addCustomIndex == -1 {
    return selections
 }

 selections = append(selections[:addCustomIndex], selections[addCustomIndex+1:]...)
 for {
    customPath, err := s.prompt.Get()
    paths = append(selections, customPath)
    another, err := s.prompt.Confirm()
    if !another {
      break
    }
  }
```

Reply:
@Lou1415926: Are you okay with leaving the code as is for now until we come up with something we're excited about?

Reply:
I'm okay with that, but before merging I have one last proposal 😂 :
```go
var results []string
var askCustom bool
for i, selection := range selections {
  if selection == statisSourceUseCustomPrompt {
    askCustom = true
    continue
  }
  results = append(results, selection)
}

 if !askCustom {
    return results
 }

 for {
    customPath, err := s.prompt.Get()
    paths = append(results, customPath)
    another, err := s.prompt.Confirm()
    if !another {
      break
    }
  }
```

I can see that the original code is trying to resize / copy the list for efficiency. However, I think the length of `selections` can't be too long, which makes the improvement in speed/memory usage fairly marginal. In this case, I feel like if we have to pick between readability & efficiency, I'd prolly vote for readability!

Reply:
Re: WX's latest suggested refactor (below), I implemented it but without `paths` because only the last custom path was being added to `results`.